Changes to prevent the composer process from exiting if it fails to fetch any OpenAPI schemas

AbhilakshSinghReen opened the pull request:
Changed the `fetchOpenApiSchemas` function in `packages/composer/lib/fetch-schemas.mjs` so that the composer CLI does not exit if it fails to fetch even a single schema.
Also, added success/error logs for fetching schema for every OpenAPI service.

Note: there were no tests for `fetchOpenApiSchemas`, so I have not changed any tests after making the above changes.

mcollina reviewed (commented):
Can you add tests? This is tested.

cc @ivan-tymoshenko

AbhilakshSinghReen commented:
> Can you add tests? This is tested.
> 
> cc @ivan-tymoshenko

Oh. When I ran `pnpm test`, c8 included this function's code in Uncovered Lines, so I assumed it was untested.

I will try to add tests and update the PR. Please give me some time.

AbhilakshSinghReen commented:
@mcollina @ivan-tymoshenko 

In the test runner script for `composer`, I see that in the `cli` folder, we're looking for files with the extension `.test.mjs`. But there are no such files in that folder.

https://github.com/platformatic/platformatic/blob/842a07b7a94c9365de0fb7432488d011f0b46b2a/packages/composer/test/runner.js#L15-L15

If I change the extension to `.test.js` I get full code coverage with c8.

Could you let me know if this is really a typo, or am I missing something?

mcollina commented:
Likely a typo

AbhilakshSinghReen commented:
> Likely a typo

Okay, so I'll correct it and change the tests to work with the new function.

AbhilakshSinghReen commented:
@mcollina 
Here is a summary of the changes:
1. Updated the test runner of `composer` to fix the typo.
2. Now, when running the tests for fetching schemas, the first test in `composer/test/cli/fetch-schema.test.js` was failing as the users schema fixture did not have parameters defined at the method level. I have updated the fixture.
3. Removed the `try-catch` block in `composer/lib/fetch-schemas.mjs:fetchOpenApiSchemas` as `Promise.allSettled` will handle the thrown errors. Also removed the corresponding test case in `composer/test/cli/fetch-schema.test.js` that was checking for the thrown error.
4. Updated the first test case in `composer/test/cli/fetch-schema.test.js` to include an invalid OpenAPI service.

We now have full test coverage for the `composer` package!

mcollina reviewed (approved):
lgtm
